### ISSUE TITLE
dartdoc throws fatal exception when package refers to another (non-pub) package

### ISSUE THREAD

**Issue by srp7474** [dart-lang/dartdoc#2378]
The context is a flutter package. 

```
Doctor summary (to see all details, run flutter doctor -v):
[?] Flutter (Channel beta, 1.22.0, on Microsoft Windows [Version 6.1.7601], locale en-CA)
[?] Android toolchain - develop for Android devices (Android SDK version 30.0.2)
[?] Chrome - develop for the web
[?] Android Studio (version 4.0)
[?] Connected device (2 available)

dartdoc version: 0.32.4
```

 When the package does not make reference to another package the dartdoc tool runs like a champ.

If the package makes reference to another package it gets an error (after deliberating for a minute or 2) as shown below.

I suspect it is a matter of setting up the dartdoc_options.yaml file correctly but I could find no doc that seemed to help.  I tried using the include / exclude options but it is not entirely clear how the name matches the package name or the library pointers.

I tried ```flutter pub get``` but it made no difference

```
dartdoc failed: NoSuchMethodError: The getter 'path' was called on null.
Receiver: null
Tried calling: path
#0      Object.noSuchMethod (dart:core-patch/object_patch.dart:51:5)
#1      _FilePackageMeta.hostedAt (package:dartdoc/src/package_meta.dart:311:53)
#2      createDartdocOptions.<anonymous closure> (package:dartdoc/src/dartdoc_options.dart:1642:38)
#3      DartdocSyntheticOption._valueAtFromSynthetic (package:dartdoc/src/dartdoc_options.dart:836:54)
#4      DartdocOptionFileSynth.valueAt (package:dartdoc/src/dartdoc_options.dart:735:12)
#5      DartdocOptionContext.linkToUrl (package:dartdoc/src/dartdoc_options.dart:1469:54)
#6      Package.documentedWhere (package:dartdoc/src/model/package.dart:193:43)
```

**Comment by srp7474** [comment]
The context is a flutter package.  When the package does not make reference to another package
the dartdoc tool runs like a champ.

If the package makes reference to another package it gets an error (after deliberating for a minute or 2) as shown below.

I suspect it is a matter of setting up the dartdoc_options.yaml file correctly but I could find no doc that seemed to help.  I tried using the include / exclude options but t is not entirely clear how the name matches the package name or the library pointers.


```
dartdoc failed: NoSuchMethodError: The getter 'path' was called on null.
Receiver: null
Tried calling: path
#0      Object.noSuchMethod (dart:core-patch/object_patch.dart:51:5)
#1      _FilePackageMeta.hostedAt (package:dartdoc/src/package_meta.dart:311:53)
#2      createDartdocOptions.<anonymous closure> (package:dartdoc/src/dartdoc_options.dart:1642:38)
#3      DartdocSyntheticOption._valueAtFromSynthetic (package:dartdoc/src/dartdoc_options.dart:836:54)
#4      DartdocOptionFileSynth.valueAt (package:dartdoc/src/dartdoc_options.dart:735:12)
#5      DartdocOptionContext.linkToUrl (package:dartdoc/src/dartdoc_options.dart:1469:54)
#6      Package.documentedWhere (package:dartdoc/src/model/package.dart:193:43)
```

**Comment by jcollins-g** [comment]
There is probably a parameter missing somewhere in the configuration, but nevertheless we shouldn't crash like this.

**Comment by srp7474** [comment]
Here is more information: I have no parameters.  Here is the dartdoc invocation
```
set flutter_root=D:\pgms\flutter-1-20\flutter
call %flutter_root%\bin\cache\dart-sdk\bin\dartdoc
```

here is the start of the 1 program 
```
library rocket_widget;
import 'package:flutter/material.dart';
import 'package:async/async.dart';
import 'package:samcas/samcas.dart';
import 'package:samcas/samwig.dart';```

**Comment by jcollins-g** [comment]
This looks like an issue where dartdoc doesn't know how to enable cross-linking between packages that aren't pub packages and isn't providing a useful error message describing where to go or even what exactly is wrong.  You can try the option `--no-link-to-remote` to disable cross linking to bypass the crash immediately.  If you want to set up remote linking, see the information on this topic in README.md.

**Comment by srp7474** [comment]
The ```--no-link-to-remote``` switch was the answer.  Ran like a champ. thanks.

**Comment by jcollins-g** [comment]
dartdoc has tests that should cover this, but they don't, because they're picking up dartdoc's linkToSource configuration in https://github.com/dart-lang/dartdoc/blob/master/dartdoc_options.yaml.  Without that, the problem reproduces easily.

**Comment by jcollins-g** [comment]
The immediate cause of the exception was fixed in #2348, however as mentioned there is a variant of the problem hiding behind that.  PR coming up.

**Comment by jcollins-g** [comment]
@srp7474 #2441 should fix the remaining issues in this space; please let me know if you can reproduce this problem after this change.
